pennylane_cirq/simulator_device.py

```diff
@@ -98,11 +108,15 @@ def apply(self, operation, wires, par):
 
             if len(basis_state_array) != len(self.qubits):
```

Comment:
Just a comment out of curiosity, but you decided to go with the `apply`, `measure()` etc. route, rather than overwrite `execute()`? I find nowadays that `apply`, `post_apply`, `measure`, etc. tend to be bit restrictive, leading to odd plugin design.

Reply:
Actually I just oriented myself at the other plugins and those mostly follow the `apply`, `measure`, etc. pattern. I think the code will become more lightweight if we move all those `BasisState` and `QubitStateVector` checks to PL core and I can remove them here.

But you're right, in the next plugin I will probably just overwrite `execute`